.github/workflows/functional.yaml

```diff
@@ -16,31 +11,47 @@ jobs:
       checks: write
     steps:
       - uses: actions/checkout@v3
-      - uses: actions/setup-python@v4.2.0
+      - uses: actions/setup-python@v4
         with:
           python-version: "3.11"
 
       - uses: aws-actions/configure-aws-credentials@v1-node16
         with:
-          role-to-assume: ${{ secrets.AWS_ROLE_ARN }}
+          role-to-assume: ${{ secrets.AWS_FTEST_ROLE }}
           aws-region: us-east-1
 
       - uses: hashicorp/setup-terraform@v2
         with:
-          terraform_version: "${{ env.terraform_version }}"
           terraform_wrapper: false
 
-      - name: Install Requirements
-        run: |
-          mkdir -p .tfcache
-          pip install --cache-dir=./.pip-cache -r requirements.txt
+      - name: Install Custodian
+        uses: ./.github/composites/install
+        with:
+          python-version: "3.11"
+
+      - name: Set up cache
+        uses: actions/cache@v3
+        id: tfcache
+        with:
+          path: .tfcache
+          key: tf-${{ runner.os }}-${{ hashFiles('**/*.tf') }}            
 
-      - name: Test
+      - name: Functional Tests
         run: |
-          pytest tests -m "not skiplive" -m terraform -n auto --junit-xml=report-aws.xml
+          mkdir -p .tfcache
+          pytest -m "terraform and audited and not skiplive" -v tests
```

Comment:
Intentional to drop the `-n auto` here?

Reply:
yeah.. there's an unrelated issue in pytest-terraform, effectively we have a race around populating the cache with providers. see https://github.com/cloud-custodian/pytest-terraform/issues/63 for deets